### DISCUSSION TITLE
Bulk edit for secrets

### DISCUSSION THREAD

**Discussion by vmatsiiako** [Infisical/infisical#5430] [Feature Request]
### Feature description
It would be great if you could click a button on a dashboard that would open the secrets in a popup with a text editor in which you are able to edit as a file. 

### Why would it be useful?
This will help users edit many secrets at once. 

### Additional context
It would be even better if this file could remember the spacing in between secrets.

**Comment by sarathkumarps17** [comment]
> ### Feature description
> It would be great if you could click a button on a dashboard that would open the secrets in a popup with a text editor in which you are able to edit as a file.
> 
> ### Why would it be useful?
> This will help users edit many secrets at once.
> 
> ### Additional context
> It would be even better if this file could remember the spacing in between secrets.

Hi @vmatsiiako, I would like to work on the UI part of this.

**Comment by Grraahaam** [comment]
Great idea for the UI, but for the CLI as well! 💻 

We could have a VSCode (or IDE agnostic) plugin that could rely on the local CLI auth that could show us a list of all our infisical projects/environments/secrets as a tmp text file that we could bulk edit easily and seamlessly from where we usually code/develop (taking advantages of the user's favorite editing tools/shortcuts). 🧩

Once edited and saved, the plugin will parse it and use the local CLI to update the corresponding projects/ens/secrets (key name + values), the same way @G3root is doing through the web-app's UI [here](https://github.com/Infisical/infisical/pull/995)

This will improve by a lot the developer's experience and interactivity, IMO!

WDYT? @maidul98 @vmatsiiako

**Comment by yonghyun-jin** [comment]
Hi, is anyone currently working on this issue? I'd love to contribute!

**Comment by akhilmhdh** [comment]
Moving this to discussions.

Dashboard now does support bulk edit - but in a more UI perspective rather than a text edit method.
